module-apps/apps-common/widgets/ProgressTimer.cpp

```diff
@@ -82,12 +82,14 @@ namespace app
 
     auto ProgressTimer::intervalReached() const noexcept -> bool
     {
-        return hasInterval && (elapsed.count() % interval.count()) == 0;
+        return hasInterval &&
+               (std::chrono::duration_cast<std::chrono::seconds>(elapsed).count() % interval.count()) == 0;
     }
 
-    void ProgressTimer::stop()
+    std::chrono::milliseconds ProgressTimer::stop()
```

Comment:
consider adding function `elapsed` instead? which will always return the current value?